Hide femr patient_ids from the user

Lalaland opened the pull request:
This PR changes the API to hide femr patient_ids from the user.

femr patient_ids still exist, as they are necessary for the C++ components and the binary format, but Python users will never see them.

ericpan64 commented on native/datasets_extension.cc:
```diff
@@ -219,29 +220,36 @@ void register_datasets_extension(py::module& root) {
             "__getitem__",
             [python_patient, pickle](py::object self_object,
-                                     uint32_t patient_id) {
+                                     uint64_t patient_id) {
```
Just curious is there a reason for the memory bump?

Lalaland replied:
OMOP person_ids (aka what we are switching to) are uint64_t. I should probably introduce a typedef to make things here much more explicit.

ericpan64 commented on native/database.cc:
```diff
@@ -781,6 +781,10 @@ uint64_t PatientDatabase::get_original_patient_id(uint32_t patient_id) {
+}
+
 boost::optional<uint32_t> PatientDatabase::get_patient_id_from_original(
     uint64_t original_patient_id) {
```
Could it be clearer to rename `get_patient_id_from_original` as something like `get_internal_id_from_original` or `get_femr_id_from_source` or some combination

Lalaland replied:
Fixed.

ericpan64 commented on tests/featurizers/test_featurizers.py:
```diff
@@ -32,7 +32,7 @@ def _assert_featurized_patients_structure(labeled_patients, featurized_patients,
     ), f"len(featurized_patients[1]) = {len(featurized_patients[1])}"
 
-    patient_ids = np.array(sorted([i for i in range(len(labeled_patients))] * len(labels_per_patient)))
+    patient_ids = np.array(sorted([i for i in labeled_patients] * len(labels_per_patient)))
```
```suggestion
    patient_ids = np.array(sorted(list(labeled_patients) * len(labels_per_patient)))
```

Are these equivalent? I don't love the suggestion since still feels kind of weird - is this an API thing or something else (like just how the test is written)?

ericpan64 commented on tests/featurizers/test_featurizers.py:
```diff
@@ -93,7 +93,7 @@ def test_count_featurizer(tmp_path: pathlib.Path):
     labeler = CodeLabeler([femr_outcome_code], time_horizon, [femr_admission_code])
 
-    patient: femr.Patient = cast(femr.Patient, database[0])
+    patient: femr.Patient = cast(femr.Patient, database[next(iter(database))])
```
If this is a common pattern - maybe implement a `database.next()` helper function for users

Lalaland replied:
This is not a common pattern. Pulling the first patient from a patient database is sorta weird.

ericpan64 reviewed (approved):
Overall lgtm! Not sure how to test the C++ code yet but added some suggestions, add as you see fit and feel free to merge

Lalaland commented:
@ericpan64 I fixed the issues you brought up and changed the API slightly. Now we have patient_offset vs patient_id, which should be more clear.

> Overall lgtm! Not sure how to test the C++ code yet 

bazel test //... in the native folder. This should be added to the CI though, let me do it after this commit is merged.
